lib/tree.rb

```diff
@@ -16,40 +16,108 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(n)
+  # Space Complexity: O(1) 
+  def add_help(current, new_node)
+    return new_node if current.nil? 
+
+    new_node.key <= current.key ?
+      current.left = add_help(current.left , new_node):
+      current.right = add_help(current.right, new_node)
+
+    return current
+  end 
+
   def add(key, value)
-    raise NotImplementedError
+    @root = add_help(@root, TreeNode.new(key,value))
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+
+  # Time Complexity: log(n)
+  # Space Complexity: O(1)
+  def help_find(current, key)
```

Comment:
👍 However since you are using recursion the space complexity is O(log n) if the tree is balanced (time as well) and O(n) if the tree is unbalanced.